ga4gh/backend.py

```diff
@@ -762,6 +770,69 @@ def runSearchDatasets(self, request):
             protocol.SearchDatasetsResponse,
             self.datasetsGenerator)
 
+    def runSearchGenotypePhenotype(self, requestStr):
+        """
+        Runs the specified SearchGenotypePhenotypeRequest.
+        We do not attempt to re-use runSearchRequest as
+        All you really need is to get the offset from the pageToken, and then
+        pass this on to the query method in the data model object.
+        """
+        # TODO verify evidence maps to drug
+
+        # validate request
+        self.startProfile()
+        try:
+            requestDict = json.loads(requestStr)
+        except ValueError:
+            raise exceptions.InvalidJsonException(requestStr)
+        requestClass = protocol.SearchGenotypePhenotypeRequest
+        responseClass = protocol.SearchGenotypePhenotypeResponse
+        self.validateRequest(requestDict, requestClass)
+        request = requestClass.fromJsonDict(requestDict)
+        if request.pageSize is None:
+            request.pageSize = self._defaultPageSize
+        if request.pageSize <= 0:
+            raise exceptions.BadPageSizeException(request.pageSize)
+        responseBuilder = protocol.SearchResponseBuilder(
+            responseClass, request.pageSize, self._maxResponseLength)
+        # ensure request has data we can query
+        if (request.evidence is None and
+                request.phenotype is None and
+                request.feature is None):
+            msg = "Error:One of evidence,phenotype or feature must be non-null"
+            raise exceptions.BadRequestException(msg)
```

Comment:
This message isn't actually returned to the user. You probably need a more specific exception, or else change the plumbing of BadRequestException to take an optional argument.